### PR TITLE
add tinycrypt to crypto

### PR DESCRIPTION
**Pull request by anjiahao1**
## Summary
add tinycrypt to crypto
and then inside of mucboot internal used lib
## Impact
mcuboot

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@gustavonihei could you take a look? This patch replace mcuboot private copy to the official tinycrypto.

File: crypto/tinycrypt/Make.defs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+ifneq ($(CONFIG_TINYCRYPT),)
+CONFIGURED_APPS += $(APPDIR)/crypto/tinycrypt
+CFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" $(APPDIR)/crypto/tinycrypt/tinycrypt/lib/include}
+CXXFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" $(APPDIR)/crypto/tinycrypt/tinycrypt/lib/include}
+endif
```
Comment (pkarashchenko) [inline crypto/tinycrypt/Make.defs:25]:
```suggestion
endif

```

File: crypto/tinycrypt/Makefile [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+include $(APPDIR)/Make.defs
+
+TINYCRYPT_URL ?= "https://github.com/intel/tinycrypt.git"
```
Comment (pkarashchenko) [inline crypto/tinycrypt/Makefile:23]: Why not download a tagged version and untar instead of cloning a master?

File: crypto/tinycrypt/Makefile [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+TINYCRYPT_UNPACK = tinycrypt
+$(TINYCRYPT_UNPACK):
+	$(Q) echo "git clone $(TINYCRYPT_URL)"
+	$(Q) git clone $(TINYCRYPT_URL)
+
+CFLAG +=-I$(TINYCRYPT_UNPACK)
```
Comment (pkarashchenko) [inline crypto/tinycrypt/Makefile:30]:
```suggestion
CFLAG += -I$(TINYCRYPT_UNPACK)
```

**Comment by gustavonihei** [comment]
> @gustavonihei could you take a look? This patch replace mcuboot private copy to the official tinycrypto.

MCUboot uses a modified version of tinycrypt, so it cannot be easily substituted.

**Comment by xiaoxiang781216** [comment]
> > @gustavonihei could you take a look? This patch replace mcuboot private copy to the official tinycrypto.
> 
> MCUboot uses a modified version of tinycrypt, so using the vanilla version from intel's repository may be not be a trivial effort.

Ok, @anjiahao1 let's keep MCUboot use the private tinycrypto version, and only move tinycrypto search path from Make.defs to Makefile.

File: crypto/tinycrypt/Makefile (hunk 3 above)
Comment (gustavonihei) [inline crypto/tinycrypt/Makefile:30]:
```suggestion
CFLAGS +=-I$(TINYCRYPT_UNPACK)
```

File: boot/mcuboot/Makefile [hunk 4]
```diff
@@ -71,6 +71,8 @@ CSRCS += $(MCUBOOT_UNPACK)/ext/tinycrypt/lib/source/aes_encrypt.c \
          $(MCUBOOT_UNPACK)/ext/tinycrypt/lib/source/ecc_dh.c \
          $(MCUBOOT_UNPACK)/ext/tinycrypt/lib/source/sha256.c \
          $(MCUBOOT_UNPACK)/ext/tinycrypt/lib/source/utils.c
+CFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" $(APPDIR)/boot/mcuboot/mcuboot/ext/tinycrypt/lib/include}
```
Comment (pkarashchenko) [inline boot/mcuboot/Makefile:74]:
```suggestion

CFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" $(APPDIR)/boot/mcuboot/mcuboot/ext/tinycrypt/lib/include}
```

**Comment by gustavonihei** [comment]
I believe the addition of `tinycrypto` requires some new information to be added to [Export Restrictions](https://github.com/apache/incubator-nuttx-apps#export-restrictions) section of the documentation.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**